### PR TITLE
[MINI-5696] Missing events for open and close dialog

### PR DESCRIPTION
**Pull request by shohiebsense**
# Description
Add onPause and onResume event on permission dialog and share dialog

## Links
[MINI-5696](https://jira.rakuten-it.com/jira/browse/MINI-5696)

## Video
https://rak.box.com/s/1pmxnhjlnc8heqi6mrm26ttmeuutoh82

# Checklist
- [x] I have read the [contributing guidelines](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md#sdk-development-learning-path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/preload/PreloadMiniAppWindow.kt [hunk 1]
```diff
@@ -40,11 +41,12 @@ class PreloadMiniAppWindow(
     @Suppress("LongParameterList")
     fun initiate(
         appInfo: MiniAppInfo?,
-        miniAppId: String,
-        versionId: String,
+        miniAppIdAndVersionIdPair: Pair<String, String>,
```
Comment (Climbatize) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/preload/PreloadMiniAppWindow.kt:44]: was this change required? If yes, should we create a more explicit dedicated data class?

**Review by Climbatize — APPROVED** [review]
Looks good

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/preload/PreloadMiniAppWindow.kt (hunk 1 above)
Comment (shohiebsense) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/preload/PreloadMiniAppWindow.kt:44]: In the sense of mistook the `first` and the `second` written by devs, yeah +1

**Review by gopi-karmakar — APPROVED** [review]

**Comment by shohiebsense** [comment]
Thanks
